### PR TITLE
Add support for uploading photos in post creation #94

### PR DESCRIPTION
**Pull request by juanbono**
### TODO: 
- [x] merge master
- [x] Make authenticated requests when uploading images.
- [x] Check file size before uploading it and compress it otherwise.

### CODE REVIEW

File: assets/js/posts_image_upload.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  Object.keys(files).forEach(i => {
+    const file = files[i];
+    const reader = new FileReader();
+
+    reader.onloadend = () => {
+      sendData(reader.result, file.type);
+    };
+
+    if (file.size > 10000000) {
+      alert("The file size must be lower than 10mb");
```
Comment (hernanex3) [inline assets/js/posts_image_upload.js]: I think is "smaller than"

File: lib/elephant_in_the_room_web/controllers/image_controller.ex [hunk 2]
```diff
@@ -20,13 +36,19 @@ defmodule ElephantInTheRoomWeb.ImageController do
     image = HTTPoison.get!(url, [], hackney: [{:follow_redirect, true}])
 
     type =
-      :proplists.get_value("Content-Type", image.headers)
+      :proplists.get_value("content-type", conn.req_headers)
       |> String.split("/")
       |> List.last()
 
-    {:ok, saved_image} = Sites.create_image(%{"binary" => image.body, "type" => type})
+    name = generate_name(type)
```
Comment (hernanex3) [inline lib/elephant_in_the_room_web/controllers/image_controller.ex]: shouldn't we check first 'content-type' to be an image? i.e. jpg or .png. I was able to upload any type file.

File: lib/elephant_in_the_room_web/templates/post/form.html.eex [hunk 3]
```diff
@@ -43,7 +43,7 @@
 
   <div class="form-group">
     <%= label f, :content, class: "uk-form-label" %>
-    <%= textarea f, :content, class: "uk-textarea uk-form-small", rows: "15"%>
+    <%= textarea f, :content, id: "post-textarea", class: "post-edit-content uk-textarea uk-form-small", rows: "15"%>
```
Comment (AminArria) [inline lib/elephant_in_the_room_web/templates/post/form.html.eex]: No need to specify an id, the default one is `post_content`

File: assets/js/posts_image_upload.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    }
+  });
+};
+
+const sendData = (data, mime) => {
+  const XHR = new XMLHttpRequest();
+  const url = "/images/binary";
+
+  XHR.onloadend = () => {
+    const postTextArea = document.getElementById("post-textarea");
```
Comment (AminArria) [inline assets/js/posts_image_upload.js]: Change `post-textarea` and all other ocurrences to `post_content`

**Comment by hernanex3** [comment]
I created an issue related to the image upload formats https://github.com/lambdaclass/elephant_in_the_room/issues/181

File: lib/elephant_in_the_room_web/controllers/image_controller.ex [hunk 5]
```diff
@@ -9,6 +9,28 @@ defmodule ElephantInTheRoomWeb.ImageController do
     |> send_resp(200, img.binary)
   end
 
+  def save_binary_image(conn, _params) do
+    {:ok, raw_body, conn} = read_body(conn)
+    IO.inspect(conn)
```
Comment (AminArria) [inline lib/elephant_in_the_room_web/controllers/image_controller.ex]: Remove this

File: lib/elephant_in_the_room_web/controllers/image_controller.ex [hunk 6]
```diff
@@ -9,6 +9,28 @@ defmodule ElephantInTheRoomWeb.ImageController do
...
+    type =
+      :proplists.get_value("content-type", conn.req_headers)
+      |> String.split("/")
+      |> List.last()
+
+    name = generate_name(type)
+
+    {:ok, saved_image} = Sites.create_image(%{"name" => name, "binary" => raw_body})
+
+    if is_image?(type) && is_small?(conn) do
```
Comment (AminArria) [inline lib/elephant_in_the_room_web/controllers/image_controller.ex]: You are making the validations **after** creating the image

**Pull request merged**